cypress/integration/access-control/history.spec.ts

```diff
@@ -0,0 +1,126 @@
+/* eslint-disable camelcase */
+import type { SBHistoryEntry, SBPodcast } from '$lib/shared/api/adapters';
+
+describe('For anonymous users', () => {
+  it('forbids reading', () => {
+    cy.createSupabaseClient().then(async (supabase) => {
+      const response = await supabase.from<SBHistoryEntry>('history').select('*').limit(1).single();
+      expect(response.status).to.be.gte(400).and.be.lessThan(500);
+    });
+  });
+
+  it('forbids creating', () => {
+    cy.createSupabaseClient().then(async (supabase) => {
+      const response = await supabase.from<SBHistoryEntry>('history').insert({ podcast_id: 1 });
+      expect(response.status).to.be.gte(400).and.be.lessThan(500);
+    });
+  });
+});
+
+describe('For authorized users', () => {
+  it('permits reading own rows', () => {
+    cy.login().then(async ([auth, supabase]) => {
+      const userID = (await auth).user?.id;
+      if (userID !== undefined) {
+        const response = await supabase.from<SBHistoryEntry>('history').select('user_id');
+        expect(response.status).to.be.gte(200).and.be.lessThan(300);
+        if (response.data !== null) {
+          for (const row of response.data) {
+            expect(row.user_id).to.be.equal(userID);
+          }
+        } else {
+          expect(response.data).not.to.be.null;
```

Comment:
Feels like it can be moved before the `if`, with a non-null assertion in the `for` loop, since it is obvious that this `expect` is never going to succeed

Reply:
Our linter is configured against non-null assertions

Reply:
Well, you can add `if (response.data === null) return;` after the `expect` just for the flow analysis, but it would still make more sense than writing an `expect` that you know for sure will fail.

Reply:
So what would you like me to do here?

Reply:
Nothing. I did it myself. Hope you don't mind)